### PR TITLE
[KeyVault] bumping azsecrets to latest azcore

### PR DESCRIPTION
**Pull request by seankane-msft**
- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

### CODE REVIEW

File: sdk/keyvault/azsecrets/go.mod [hunk 1]
```diff
@@ -2,9 +2,13 @@ module github.com/Azure/azure-sdk-for-go/sdk/keyvault/azsecrets
...
 go 1.16
 
+replace github.com/Azure/azure-sdk-for-go/sdk/internal => ../../internal
+
+replace github.com/Azure/azure-sdk-for-go/sdk/azidentity => ../../azidentity
+
 require (
-	github.com/Azure/azure-sdk-for-go/sdk/azcore v0.19.0
+	github.com/Azure/azure-sdk-for-go/sdk/azcore v0.20.0
 	github.com/Azure/azure-sdk-for-go/sdk/azidentity v0.11.0
```
Comment (jhendrixMSFT) [inline sdk/keyvault/azsecrets/go.mod:11]: I assume this will wait for the upcoming v0.12.0 release.

File: sdk/keyvault/azsecrets/go.mod [hunk 2]
```diff
@@ -2,9 +2,13 @@ module github.com/Azure/azure-sdk-for-go/sdk/keyvault/azsecrets
 
 go 1.16
 
+replace github.com/Azure/azure-sdk-for-go/sdk/internal => ../../internal
```
Comment (jhendrixMSFT) [inline sdk/keyvault/azsecrets/go.mod:5]: This can be removed now yes?

File: sdk/keyvault/azsecrets/go.mod (hunk 1 above)
Comment (seankane-msft) [reply sdk/keyvault/azsecrets/go.mod:11]: Yes, after the azidentity release I can bump to v0.12.0 and remove that directive as well.

File: sdk/keyvault/azsecrets/CHANGELOG.md [hunk 3]
```diff
@@ -9,6 +9,7 @@
 ### Bugs Fixed
 
 ### Other Changes
+* Bumps `azcore` dependency to `v0.20.0` and `azidentity` to `v0.11.0`
```
Comment (chlowell) [inline sdk/keyvault/azsecrets/CHANGELOG.md:12]:
```suggestion
* Bumps `azcore` dependency to `v0.20.0` and `azidentity` to `v0.12.0`
```

File: sdk/keyvault/azsecrets/client.go [hunk 4]
```diff
@@ -67,12 +50,21 @@ func NewClient(vaultUrl string, credential azcore.TokenCredential, options *Clie
 		options = &ClientOptions{}
 	}
 
-	conn := internal.NewConnection(credential, options.toConnectionOptions())
+	options.PerRetryPolicies = append(
```
Comment (chlowell) [inline sdk/keyvault/azsecrets/client.go:53]: We shouldn't mutate options passed by applications (what happens when an application passes the same options pointer in multiple calls)

File: sdk/keyvault/azsecrets/autorest.md [hunk 5]
```diff
@@ -8,12 +8,17 @@ These settings apply only when `--go` is specified on the command line.
 ``` yaml
 go: true
 version: "^3.0.0"
-input-file: https://github.com/Azure/azure-rest-api-specs/blob/1e2c9f3ec93078da8078389941531359e274f32a/specification/keyvault/data-plane/Microsoft.KeyVault/stable/7.2/secrets.json
+input-file:
+- https://github.com/Azure/azure-rest-api-specs/blob/ecdce42924ed0f7e60a32c74bc0eb674ca6d4aae/specification/keyvault/data-plane/Microsoft.KeyVault/stable/7.2/common.json
+- https://github.com/Azure/azure-rest-api-specs/blob/ecdce42924ed0f7e60a32c74bc0eb674ca6d4aae/specification/keyvault/data-plane/Microsoft.KeyVault/stable/7.2/secrets.json
+- https://github.com/Azure/azure-rest-api-specs/blob/ecdce42924ed0f7e60a32c74bc0eb674ca6d4aae/specification/keyvault/data-plane/Microsoft.KeyVault/stable/7.2/rbac.json
+- https://github.com/Azure/azure-rest-api-specs/blob/ecdce42924ed0f7e60a32c74bc0eb674ca6d4aae/specification/keyvault/data-plane/Microsoft.KeyVault/stable/7.2/securitydomain.json
```
Comment (chlowell) [inline sdk/keyvault/azsecrets/autorest.md:15]: Aren't security domains relevant only to MHSM?

File: sdk/keyvault/azsecrets/CHANGELOG.md [hunk 6]
```diff
@@ -1,6 +1,6 @@
 # Release History
 
-## 0.1.2 (Unreleased)
+## 0.2.0 (Unreleased)
```
Comment (chlowell) [inline sdk/keyvault/azsecrets/CHANGELOG.md:3]: You could sneak the date in here as well

Comment (seankane-msft) [reply sdk/keyvault/azsecrets/CHANGELOG.md:3]: I'm going to let the Prepare-Release.ps1 script take care of this

File: sdk/keyvault/azsecrets/autorest.md (hunk 5 above)
Comment (seankane-msft) [reply sdk/keyvault/azsecrets/autorest.md:15]: Yes but these are the files I found I needed to have all of the models for the generated code. If there's a different combination you know of I'll remove them in a separate PR

**Review by chlowell — APPROVED** [review]

**Pull request merged**